packages/nimble-blazor/Tests/NimbleBlazor.Tests/Unit/Components/NimbleSpinnerTests.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Linq.Expressions;
+using Bunit;
+using Xunit;
+
+namespace NimbleBlazor.Tests.Unit.Components;
+
+/// <summary>
+/// Tests for <see cref="NimbleSpinner"/>.
+/// </summary>
+public class NimbleSpinnerTests
+{
+    [Fact]
+    public void NimbleSpinner_Render_HasSpinnerMarkup()
+    {
+        var context = new TestContext();
+        context.JSInterop.Mode = JSRuntimeMode.Loose;
+        var expectedMarkup = "nimble-spinner";
+
+        var spinner = context.RenderComponent<NimbleSpinner>();
+
+        Assert.Contains(expectedMarkup, spinner.Markup);
+    }
+
+    [Fact]
+    public void NimbleSpinner_SupportsAdditionalAttributes()
+    {
+        var context = new TestContext();
+        context.JSInterop.Mode = JSRuntimeMode.Loose;
+        var exception = Record.Exception(() => context.RenderComponent<NimbleSpinner>(ComponentParameter.CreateParameter("class", "foo")));
+        Assert.Null(exception);
+    }
+
+    [Theory]
+    [InlineData(SpinnerAppearance.Default, "<nimble-spinner((?!appearance).)*>")]
```

Comment:
While I appreciate the rigor here, I think it's sufficient to follow our other patterns for the default case, where we just check against the literal `<nimble-spinner>`. A default component will never have other attributes set in it, so this is safe to do. It also allows a test reader to not have to be familiar with regular expressions to know what we're testing.